nilearn/input_data/nifti_labels_masker.py

```diff
@@ -278,7 +279,10 @@ def transform_single_imgs(self, imgs, confounds=None):
         if not hasattr(self, '_resampled_mask_img'):
             self._resampled_mask_img = self.mask_img_
         if self.resampling_target == "data":
-            imgs_ = _utils.check_niimg_4d(imgs)
+            try:
+                imgs_ = _utils.check_niimg_4d(imgs)
+            except:
```

Comment:
I would catch the specific error you expect here (`DimensionError`).
Besides, I think using `check_niimg` would take care of that for you:

```python
imgs_ = _utils.check_niimg(imgs)
```
If you want to enforce 4D, you can use:

```python
imgs_ = _utils.check_niimg(imgs, atleast_4d=True)
```

Reply:
Hello Nicolas, 
you mean the check_niimg is able to recognize the dimensionality (3d or 4d) or the try-catch is still needed?